src/core/interpreter.cc

```diff
@@ -1104,4 +1108,14 @@ void InterpreterManager::Reset() {
   VLOG(1) << "InterpreterManager::Reset ended";
 }
 
+void InterpreterManager::Alter(std::function<void(Interpreter*)> modf) {
+  vector<Interpreter*> taken;
```

Comment:
why not 
```
for (Interpreter* ir : available_) {
  modf(ir);
}
```
with fiberatomic  guard to make sure modf does not preempt?

Reply:
please add a comment here explaining that cb can preempt so we swap first and then return to avoid data races.

Reply:
> with fiberatomic guard to make sure modf does not preempt?

My whole idea was to add support for preemtions to avoid fully freezing

Reply:
But tell me if it's not needed, I'll remove it

Reply:
I do not have strong opinion 😄 , let's keep it with the comment